admin-dev/themes/new-theme/js/components/form/form-object-mapper.js

```diff
@@ -0,0 +1,282 @@
+    // acknowledges the update
+    this.updateModelEventName = inputConfig.updateModel || 'updateModel';
+
+    // This event is emitted each time the object is updated (from both input change and external event)
+    this.modelUpdatedEventName = inputConfig.modelUpdated || 'modelUpdated';
+    // This event is emitted each time an object field is updated (from both input change and external event)
+    this.modelFieldUpdatedEventName = inputConfig.modelFieldUpdated || 'modelFieldUpdated';
+
+    this.modelMapping = {};
+    this.formMapping = {};
+    this.object = {};
+
+    this.initFormMapping();
+    this.watchUpdates();
+    this.updateFullObject();
+  }
+
+  /**
+   * Returns the object mapped to the form (current live state)
+   *
+   * @returns {*|{}}
+   */
+  getObject() {
+    return this.object;
+  }
+
+  /**
+   * Get a field from the object based on the model key
+   *
+   * @param modelKey (string)
+   *
+   * @returns {*|{}}
+   */
+  get(modelKey) {
+    const modelKeys = modelKey.split('.');
+
+    return $.serializeJSON.deepGet(this.object, modelKeys);
+  }
+
+  /**
+   * Returns the input associated to a model field (in case the field is mapped to
+   * several inputs the default one, first configured, is always used).
+   *
+   * @param modelKey {string}
+   * @returns {undefined|jQuery}
+   */
+  getInput(modelKey) {
+    if (!Object.prototype.hasOwnProperty.call(this.modelMapping, modelKey)) {
+      return undefined;
+    }
+
+    return $(`[name="${this.modelMapping[modelKey]}"]`);
+  }
+
+  /**
+   * Set a value to a field of the object based on the model key, the object itself is updated
+   * of course but the mapped inputs are also synced (all of them if multiple). Events are also
+   * triggered to indicate the object has been updated (the general and the individual field ones).
+   *
+   * @param modelKey {string}
+   * @param value {*|{}}
+   */
+  set(modelKey, value) {
+    if (!Object.prototype.hasOwnProperty.call(this.modelMapping, modelKey) || value === this.get(modelKey)) {
+      return;
+    }
+
+    // First update the inputs then the model, so that the event is sent at last
+    this.updateInputValue(modelKey, value);
+    this.updateObjectByKey(modelKey, value);
+    this.eventEmitter.emit(this.modelUpdatedEventName, this.object);
+  }
+
+  /**
+   * Watches if changes happens from the form or via an event.
+   */
+  watchUpdates() {
+    this.$form.on('keyup change dp.change', ':input', _.debounce(
+      (event) => this.inputUpdated(event),
+      200,
+      {maxWait: 1000},
+    ));
+    this.eventEmitter.on(this.updateModelEventName, () => this.updateFullObject());
+  }
+
+  /**
+   * Triggered when a form input has been changed.
+   *
+   * @param event {Object}
+   */
+  inputUpdated(event) {
+    const target = event.currentTarget;
+
+    if (!Object.prototype.hasOwnProperty.call(this.formMapping, target.name)) {
+      return;
```

Comment:
It looks pretty silent, don't we want to log it? For development purpose mainly

Reply:
I agreed with you at first, but then I realized this listener os called for any modifications in the whole form: `this.$form.on('keyup change dp.change', ..)` But not all the inputs are necessarily configured in the form mapping

For now only a few are handled, so any modifs on other fields would trigger a `console.log` which can be quite bothersome in the end no?

Reply:
Yeah but if you want to use the component for only a part of the form (it only handles the configuration that is set) Then we will have a lot of noisy logs for all the inputs that are not handled